### PR TITLE
Only allow using Teleportation Potion and friends whilst holding them

### PR DESCRIPTION
**Pull request by drunderscore**
Some simple protections against using these when you shouldn't be able to!

### CODE REVIEW

File: TShockAPI/GetDataHandlers.cs [hunk 1]
```diff
@@ -3814,20 +3814,41 @@ void Fail(string tpItem)
 			switch (type)
 			{
 				case 0: // Teleportation Potion
+					if (args.Player.ItemInHand.type != ItemID.TeleportationPotion &&
+					    args.Player.SelectedItem.type != ItemID.TeleportationPotion)
```
Comment (QuiCM) [inline TShockAPI/GetDataHandlers.cs:3818]: If I consume my last potion to teleport, leaving the slot empty, does this check pass or fail?

Comment (drunderscore) [reply TShockAPI/GetDataHandlers.cs:3818]: This check passes, surprisingly. Good edge-case thinking though!

Comment (QuiCM) [reply TShockAPI/GetDataHandlers.cs:3818]: That's neat. Genuinely surprised lol

**Review by QuiCM — DISMISSED** [review]

File: TShockAPI/GetDataHandlers.cs (hunk 1 above)
Comment (drunderscore) [reply TShockAPI/GetDataHandlers.cs:3818]: Yeah sorta lucked out there... wouldn't bet on the packet order for other consumables being action then item removal though!

**Review by hakusaro — APPROVED** [review]

**Pull request merged**